### ISSUE TITLE
LayerNorm does not normalize its input correctly

### ISSUE THREAD

**Issue by booti386** [tracel-ai/burn#2185]
**Describe the bug**
Both from pytorch documentation...:
https://pytorch.org/docs/stable/generated/torch.nn.LayerNorm.html#torch.nn.LayerNorm
```
y = (x − E[x]) / sqrt(Var[x] + ϵ)∗γ + β
```

... and from pytorch ATen implementation...:
https://github.com/pytorch/pytorch/blob/6564e746ed494505be5f65aa99f8cf1884bb6342/aten/src/ATen/native/cpu/layer_norm_kernel.cpp#L115
```python
  rstd_val = float(1) / std::sqrt(rstd_val + eps)
```

... they normalize by `sqrt(var + eps)`. Now if we take a look at burn...:
https://github.com/tracel-ai/burn/blob/c29ed43441fc5b3286749816c9069ed90f178a77/crates/burn-core/src/nn/norm/layer.rs#L70

... we normalize by `sqrt(var) + eps`, which leads to precision issues.
